pkg/pkg.mk

```diff
@@ -9,8 +9,8 @@ PKG_BUILDDIR?=$(BINDIRBASE)/pkg/$(BOARD)/$(PKG_NAME)
 git-download: $(PKG_BUILDDIR)/.git-downloaded
 
 GIT_APPLY_PATCHES:=if test -d "$(PKG_DIR)"/patches; then \
-	git -C "$(PKG_BUILDDIR)" am --ignore-whitespace "$(PKG_DIR)"/patches/*.patch; \
-	fi
+    cd "$(PKG_BUILDDIR)" && git am --ignore-whitespace "$(PKG_DIR)"/patches/*.patch; \
+  fi
```

Comment:
That is a correct statement about how *make* handles invocation.
I'm talking about the code block, which is bound to `GIT_APPLY_PATCHES`. It leaves you in `PKG_BUILDDIR` after it is invoked. Given that, it has a side effect and you can't just hope that this effect is reverted later by magic.
If it is used as in line 26 then for the expression in line 27 *CWD* is `PKG_BUILDDIR` (Same thing as line 24-25). If PKG_BUILDDIR is not absolute (a precondition you have to test/assure if it exist) then line 27 can fail.
Funny enough... Even if it fails, you will not notice it until "strange things" happen (See comment to line 28).